### ISSUE TITLE
uds (flexvol) "no such file or directory" regression in 3.24.0 from 3.23.3

### ISSUE THREAD

**Issue by fernferret** [projectcalico/calico#6590]
## Expected Behavior
`kube-controller-manager` shouldn't log errors about `Error dynamically probing plugins`.

A fix for #5356 was accepted in #5515, which was released into `v1.23.0`. This regressed in #6539 causing the error to return in `v1.24.0`, since the `flexvol` binary is no longer built statically.

This at least affects vanilla Kubernetes 1.23.x, I am unsure about other flavors but it would likely depend on if the `kube-controller-manager` image had the dependencies in it.

## Current Behavior

The `uds` aka `flexvol` binary is no longer statically linked which causes the following error to come out of the `kube-controller-manager` container:

```
E0820 09:42:23.066066       1 plugins.go:752] "Error dynamically probing plugins" err="error creating Flexvolume plugin from directory nodeagent~uds, skipping. Error: unexpected end of JSON input"
W0820 09:42:23.066036       1 driver-call.go:149] FlexVolume: driver call failed: executable: /usr/libexec/kubernetes/kubelet-plugins/volume/exec/nodeagent~uds/uds, args: [init], error: fork/exec /usr/libexec/kubernetes/kubelet-plugins/volume/exec/nodeagent~uds/uds: no such file or directory, output: ""
```

## Possible Solution

I see that the builder change was made to support FIPS, so I'm not sure what can be done to maintain compliance and build a static image.

## Steps to Reproduce (for bugs)
```
% docker create --name flexvol3.23.3 docker.io/calico/pod2daemon-flexvol:v3.23.3
7e24c369ffdfcb5314bbff4104f0f44b682d7605117a458122c012f033fa8c99
% docker create --name flexvol3.24.0 docker.io/calico/pod2daemon-flexvol:v3.24.0
ec54abce1fedd3491f06a3998074f7e0cc8215c6a58c59cc13cfdf93a34c321c

% docker cp flexvol3.23.3:/usr/local/bin/flexvol flexvol3.23.3
% docker cp flexvol3.24.0:/usr/local/bin/flexvol flexvol3.24.0

% ldd flexvol3.23.3
	not a dynamic executable
% ldd flexvol3.24.0
	linux-vdso.so.1 (0x00007ffc3efe4000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fcc5a7f0000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fcc5a5fe000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fcc5a81d000)
```

## Context

I was trying to get rid of this error message and finally did last week in `v1.23.x`, only for it to come back in `v1.24.0`

## Your Environment

* Calico version: `v1.24.0`
* Orchestrator version (e.g. kubernetes, mesos, rkt): `Kubernetes v1.23.9`
* Operating System and version: `Ubuntu 20.04.4`

**Comment by lwr20** [comment]
@caseydavenport knows about FIPS and @mgleung knows about flexvol, so pinging them

**Comment by rene-dekker** [comment]
My bad.
This PR should disable cgo for flexvol while keeping it for the other images in this folder: https://github.com/projectcalico/calico/pull/6600

**Comment by caseydavenport** [comment]
Looks like this was fixed and should be available in v3.24.1 very soon.
